File: src/jk_5/nailed/map/MappackInitializationException.java
package jk_5.nailed.map;

/**
 * No description given
 *
 * @author jk-5
 */
public class MappackInitializationException extends RuntimeException {

    public MappackInitializationException(Mappack pack) {
        super("Error while reading mappack " + pack.getInternalName());
    }

    public MappackInitializationException(Mappack pack, Throwable t) {
        super("Error while reading mappack " + pack.getInternalName(), t);
    }

    public MappackInitializationException(Mappack pack, String msg) {
        super("Error while reading mappack " + pack.getInternalName() + ": " + msg);
    }

    public MappackInitializationException(Mappack pack, String msg, Throwable t) {
        super("Error while reading mappack " + pack.getInternalName() + ": " + msg, t);
    }
}

File: src/main/scala/jk_5/nailed/map/MappackInitializationException.scala
package jk_5.nailed.map

/**
 * No description given
 *
 * @author jk-5
 */
class MappackInitializationException(msg: String, cause: Throwable = null) extends RuntimeException(msg, cause) {
  def this(pack: Mappack) = this("Error while reading mappack " + pack.internalName)
  def this(pack: Mappack, t: Throwable)  = this("Error while reading mappack " + pack.internalName, t)
  def this(pack: Mappack, msg: String) = this("Error while reading mappack " + pack.internalName + ": " + msg)
  def this(pack: Mappack, msg: String, t: Throwable) = this("Error while reading mappack " + pack.internalName + ": " + msg, t)
}